app/main.py

```diff
@@ -0,0 +1,217 @@
+            return
+
+        log.info("Looking for files...")
+        log.info(vars(self.args))
+        worker = FTPWorker(**self.ftp_params(self.services))
+        encrypter = Encrypter(**self.encrypt_params(self.services, locn=self.args.keys))
+        with worker as active:
+            for job in active.get(active.filenames):
+                if job.fn not in self.recent:
+                    data = job._asdict()
+                    data["contents"] = base64.standard_b64encode(job.contents).decode("ascii")
+                    data["ts"] = job.ts.isoformat()
+                    payload = encrypter.encrypt(data)
+                    msg_id = self.publisher.publish_message(payload)
+                    self.recent[job.fn] = (job.ts, msg_id)
+
+            now = datetime.datetime.utcnow()
+            for fn, (ts, msg_id) in self.recent.copy().items():
+                if msg_id not in self.publisher._deliveries:
+                    active.delete(fn)
+                if now - ts > datetime.timedelta(hours=1):
+                    del self.recent[fn]
+
+
+def make_app():
+    return tornado.web.Application([
+        (r"/recent", StatusService, {"work": Task}),
+    ])
+
+
+def parser(description="SEFT Publisher service."):
+    here = os.path.dirname(__file__)
+    p = argparse.ArgumentParser(description)
+    p.add_argument(
+        "-v", "--verbose", required=False,
+        action="store_const", dest="log_level",
+        const=logging.DEBUG, default=logging.INFO,
+        help="Increase the verbosity of output")
+    p.add_argument(
+        "--keys", default=os.path.abspath(os.path.join(here, "test")),
+        help="Set a path to the keypair directory.")
+    p.add_argument(
+        "--port", type=int, default=int(os.getenv("PORT", "8080")),
+        help="Set a port for the service.")
+    p.add_argument(
+        "--test", default=False, action="store_true",
+        help="Configure for functional test.")
+    p.add_argument(
+        "--log", default=None, dest="log_path",
+        help="Set a file path for log output")
+    return p
+
+
+def configure_log(args, name="sdx.seft"):
+    log = logging.getLogger(name)
+    log.setLevel(args.log_level)
+
+    formatter = logging.Formatter(
+        "%(asctime)s %(levelname)-7s %(name)s|%(message)s")
+    ch = logging.StreamHandler()
+
+    if args.log_path is None:
+        ch.setLevel(args.log_level)
+    else:
+        fh = WatchedFileHandler(args.log_path)
+        fh.setLevel(args.log_level)
+        fh.setFormatter(formatter)
+        log.addHandler(fh)
+        ch.setLevel(logging.WARNING)
+
+    ch.setFormatter(formatter)
+    log.addHandler(ch)
+    log = logging.getLogger("pika")
+    log.setLevel(args.log_level)
+    log.addHandler(ch)
+    return name
+
+
+def main(args):
+    log = logging.getLogger(configure_log(args))
+    log.info("Launched in {0}.".format(os.getcwd()))
+
+    services = json.loads(os.getenv("VCAP_SERVICES", "{}"))
```

Comment:
This is specific to getting config from CF. I appreciate you're using CF for testing but production code will need  to be configured from environment variables as the target environment is NGD.

Therefore we should remove this line.